src/pages/ShareMessagePage.js

```diff
@@ -9,27 +9,27 @@ import ScreenWrapper from '../components/ScreenWrapper';
 import TextInput from '../components/TextInput';
 import withLocalize, {withLocalizePropTypes} from '../components/withLocalize';
 // import additionalAppSetup from './src/setup';
-import OptionRowLHN from '../components/LHNOptionsList/OptionRowLHN';
+import CONST from '../CONST';
+import OptionRowLHNData from '../components/LHNOptionsList/OptionRowLHNData';
 import Navigation from '../libs/Navigation/Navigation';
 import * as Report from '../libs/actions/Report';
 import styles from '../styles/styles';
 
 function ShareMessagePage(props) {
     const reportID = props.route.params.reportID;
-    const [attachment, setAttachment] = useState(props.route.params.share);
-    const [message, setMessage] = useState('');
+    const attachment = props.route.params.share;
+    const source = attachment.data;
+    const uri = attachment.data;
+    const name = source.split('/').pop();
+    const type = attachment.mimeType;
+    const isTextShare = type === 'text/plain';
 
-    useEffect(() => {
-        if (Platform.OS === 'ios') {
-            ShareMenuReactView.data().then((share) => setAttachment(share && share.data[0]));
-        } else {
-            setAttachment(props.route.params.share);
-        }
-    }, [props.route.params.share]);
+    const [message, setMessage] = useState(isTextShare ? source : '');
 
-    // TODO: fix these
-    const close = Platform.select({ios: () => ShareMenuReactView.dismissExtension(), default: () => Navigation.dismissModal()});
-    const goBack = Platform.select({ios: () => console.log('TODO: fix goBack'), default: () => Navigation.goBack()});
+    const dismiss = Platform.select({
```

Comment:
should libify this with the platform extensions so we can avoid the Platform.select